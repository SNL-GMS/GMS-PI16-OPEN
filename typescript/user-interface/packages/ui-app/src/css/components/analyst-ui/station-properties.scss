.station-properties-display-window {
  height: 100%;
  width: 100%;
  padding: 0.5;
  display: flex;
  justify-content: center;
  align-items: center;
}

.station-properties-panel {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: column;
  overflow-y: auto;
  overflow-x: hidden;
}

.station-properties-table {
  height: 100%;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  // wrapper wraps a single table
  &__wrapper {
    height: 100%;
    flex-grow: 1;
    flex-basis: 0;
  }
}

.channel-configuration-table {
  padding-bottom: 0;
}

//AG grid seems to start row count at zero which is considered even
.ag-theme-dark .ag-row-even {
  background-color: $gms-prominent-background;
}

.ag-theme-dark .ag-row-odd {
  background-color: $gms-popover-background;
}

.ag-theme-dark .ag-ltr .ag-cell {
  border-right-style: solid;
  border-right-color: #182026;
  border-right-width: 2px;
}

.station-properties-table-cell {
  position: relative;
  width: 100%;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding: 0 10px;
  border-style: solid;
  border: #3781e1;
  font-size: 16px;
  font-weight: 400;
  color: $core-main;
  /// Table Value styles
  .station-properties-table-cell__value {
    position: relative;
    z-index: 1;
    display: flex;
    width: 100%;
    align-items: center;
    color: $core-main;
    & span {
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }
    &--numeric {
      justify-content: flex-end;
      flex-grow: 10;
      width: auto;
    }
    &--center {
      justify-content: center;
      flex-grow: 10;
      width: auto;
    }
  }

  .station-properties-table-cell-container {
    padding: 2px 0;
    background-color: $core-transparent;
  }

  /// Selection
  @mixin selectedBorder($where...) {
    position: relative;
    &::after {
      @include fillAllSpaceAbsolutely;
      content: '';
      @if length($where) > 0 {
        @each $place in $where {
          border-#{$place}: 4px solid lighten($core-input-highlight, 10%);
        }
      } @else {
        border: 4px solid lighten($core-input-highlight, 10%);
      }
      z-index: 1;
      opacity: 1;
      background-color: transparent;
    }
  }

  .is-selected & {
    @include selectedBorder('top', 'bottom', 'left', 'right');
  }
  .is-selected &,
  .ag-row-selected & {
    border: transparent !important;
  }
}

@mixin headerFontStyle {
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 1px;
  padding: 0 10px;
}

.station-properties-header-cell {
  @include headerFontStyle;
  color: $gms-main;

  &.ag-header-cell {
    padding: 0 10px;
  }
}
