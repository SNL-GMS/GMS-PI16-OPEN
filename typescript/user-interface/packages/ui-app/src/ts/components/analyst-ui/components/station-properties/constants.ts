export const headerCellBlockClass = 'station-properties-header-cell';
export const smallCellWidthPx = 105;
export const medCellWidthPx = 200;
export const largeCellWidthPx = 300;
export const INVALID_CELL_TEXT = 'Unknown';
