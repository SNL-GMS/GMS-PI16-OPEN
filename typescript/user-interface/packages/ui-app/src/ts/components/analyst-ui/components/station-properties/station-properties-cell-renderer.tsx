/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import * as StringUtils from '@gms/common-util/lib/util/string-util';
import { classList, getDataAttributesFromProps } from '@gms/ui-util';
import * as React from 'react';

import { StationPropertiesCellRendererProps } from './types';

export interface StationPropertiesTableCellRendererProps {
  className?: string;
  heightCSS?: string;
  isNumeric?: boolean;
  shouldCenterText?: boolean;
  tooltipMsg?: string;
  value?: string;
  leftChild?: JSX.Element | React.FunctionComponent | React.ComponentClass;
}

/**
 * Cell renderer for table cells. Accepts classes,
 * data-attributes, and a flag to indicate if it is numeric
 */
export const StationPropertiesTableCellRendererComponent: React.FunctionComponent<React.PropsWithChildren<
  StationPropertiesTableCellRendererProps
>> = props => {
  const dataAttributes = getDataAttributesFromProps(props);
  return (
    <div
      style={{
        height: props.heightCSS ? props.heightCSS : '36px',
        overflow: 'hidden'
      }}
      data-cy="station-properties-table-cell"
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...dataAttributes}
      className={`station-properties-table-cell ${props.className || ''}`}
      title={props.tooltipMsg ? props.tooltipMsg : props.value}
    >
      {props.leftChild}
      {props.value ? (
        <div
          className={classList(
            {
              'station-properties-table-cell__value--numeric': props.isNumeric,
              'station-properties-table-cell__value--center': props.shouldCenterText
            },
            'station-properties-table-cell__value'
          )}
          data-cy="station-properties-table-cell__value"
        >
          <span>{props.value}</span>
        </div>
      ) : null}
      {props.children}
    </div>
  );
};

export const getColumnPosition = (
  props: StationPropertiesCellRendererProps
): number | 'first' | 'last' => {
  const index = props.columnApi
    .getAllDisplayedColumns()
    .findIndex(c => c.getColId() === props.colDef.colId);
  let retValue: number | 'first' | 'last' = index;
  if (index === 0) {
    retValue = 'first';
  } else if (index === props.columnApi.getAllDisplayedColumns().length - 1) {
    retValue = 'last';
  }
  return retValue;
};

export const StationPropertiesCellRenderer: React.FunctionComponent<StationPropertiesCellRendererProps> = (
  props: StationPropertiesCellRendererProps
) => {
  const { valueFormatted, value } = props;
  return (
    <StationPropertiesTableCellRendererComponent
      data-col-position={getColumnPosition(props)}
      value={valueFormatted ?? value}
      isNumeric={StringUtils.isNumeric(value)}
    />
  );
};
