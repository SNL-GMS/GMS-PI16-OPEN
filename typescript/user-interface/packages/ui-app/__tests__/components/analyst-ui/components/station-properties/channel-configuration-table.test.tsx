import {
  ChannelTypes,
  CommonTypes,
  ProcessingStationTypes,
  ResponseTypes,
  StationTypes
} from '@gms/common-model';
import { toEpochSeconds } from '@gms/common-util';
import { mount } from 'enzyme';
import * as React from 'react';

import { ChannelConfigurationTable } from '../../../../../src/ts/components/analyst-ui/components/station-properties/channel-configuration-table';
import { channelColumnsToDisplay } from '../../../../../src/ts/components/analyst-ui/components/station-properties/station-properties-utils';

window.ResizeObserver = jest.fn(() => {
  return { observe: jest.fn(), disconnect: jest.fn(), unobserve: jest.fn() };
});
const globalAny: any = global;
globalAny.ResizeObserver = window.ResizeObserver;
globalAny.DOMRect = jest.fn(() => ({}));

describe('Channel Configuration Table', () => {
  const station: StationTypes.Station = {
    name: 'STA',
    effectiveAt: toEpochSeconds('2021-04-20T16:11:31.118870Z'),
    effectiveUntil: toEpochSeconds('2021-04-20T16:11:31.118870Z'),
    relativePositionsByChannel: new Map<string, ChannelTypes.RelativePosition>([
      [
        'Real Channel Name One',
        {
          northDisplacementKm: 50.0,
          eastDisplacementKm: 5.0,
          verticalDisplacementKm: 10.0
        }
      ],
      [
        'Real Channel Name Two',
        {
          northDisplacementKm: 50.0,
          eastDisplacementKm: 5.0,
          verticalDisplacementKm: 10.0
        }
      ]
    ]),
    channelGroups: [{} as any],
    allRawChannels: [{} as any],
    description: 'This is a test station',
    location: {
      latitudeDegrees: 35.647,
      longitudeDegrees: 100.0,
      depthKm: 50.0,
      elevationKm: 10.0
    },
    type: StationTypes.StationType.SEISMIC_1_COMPONENT
  };
  const channels: ChannelTypes.Channel[] = [
    {
      station: { name: 'STA' },
      response: {
        fapResponse: ({
          // TODO no id currently in FAP response
          id: '966deee1-545e-44ed-a346-b029ab111151'
        } as unknown) as ResponseTypes.FrequencyAmplitudePhase,
        calibration: {
          calibrationPeriodSec: 14.5,
          calibrationTimeShift: 'PT2S',
          calibrationFactor: {
            value: 1.2,
            standardDeviation: 0.112,
            units: CommonTypes.Units.SECONDS
          }
        },
        effectiveAt: toEpochSeconds('2021-04-20T16:11:30.674083Z'),
        effectiveUntil: toEpochSeconds('2021-04-20T16:11:30.674083Z'),
        id: '5dc3dce4-ae3c-3fb0-84ba-2f83fb5a89d2'
      },
      processingMetadata: new Map<ChannelTypes.ChannelProcessingMetadataType, any>([
        [ChannelTypes.ChannelProcessingMetadataType.CHANNEL_GROUP, 'CHANNEL_GROUP']
      ]),
      channelBandType: ChannelTypes.ChannelBandType.BROADBAND, // 'BROADBAND',
      channelInstrumentType: ChannelTypes.ChannelInstrumentType.HIGH_GAIN_SEISMOMETER,
      channelOrientationType: ChannelTypes.ChannelOrientationType.VERTICAL,
      channelOrientationCode: 'Z',
      channelDataType: ProcessingStationTypes.ChannelDataType.DIAGNOSTIC_SOH,
      nominalSampleRateHz: 66.0,
      orientationAngles: {
        horizontalAngleDeg: 65.0,
        verticalAngleDeg: 135.0
      },
      configuredInputs: [],
      processingDefinition: new Map<string, any>(),
      description: 'Example description',
      canonicalName: 'Canonical Name One',
      location: {
        latitudeDegrees: 35.0,
        longitudeDegrees: -125.0,
        depthKm: 100.0,
        elevationKm: 5500.0
      },
      units: CommonTypes.Units.NANOMETERS,
      effectiveAt: toEpochSeconds('1970-01-01T00:00:00Z'),
      effectiveUntil: toEpochSeconds('1970-01-01T00:00:00Z'),
      name: 'Real Channel Name One'
    }
  ];
  const wrapper = mount(
    <ChannelConfigurationTable
      channels={channels}
      stationData={station}
      columnsToDisplay={channelColumnsToDisplay}
    />
  );
  test('can mount', () => {
    expect(ChannelConfigurationTable).toBeDefined();
  });
  test('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
