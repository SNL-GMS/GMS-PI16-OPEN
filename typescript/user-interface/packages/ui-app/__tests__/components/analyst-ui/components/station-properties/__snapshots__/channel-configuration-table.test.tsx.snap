// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel Configuration Table matches snapshot 1`] = `
<ChannelConfigurationTable
  channels={
    Array [
      Object {
        "canonicalName": "Canonical Name One",
        "channelBandType": "B",
        "channelDataType": "DIAGNOSTIC_SOH",
        "channelInstrumentType": "H",
        "channelOrientationCode": "Z",
        "channelOrientationType": "Z",
        "configuredInputs": Array [],
        "description": "Example description",
        "effectiveAt": 0,
        "effectiveUntil": 0,
        "location": Object {
          "depthKm": 100,
          "elevationKm": 5500,
          "latitudeDegrees": 35,
          "longitudeDegrees": -125,
        },
        "name": "Real Channel Name One",
        "nominalSampleRateHz": 66,
        "orientationAngles": Object {
          "horizontalAngleDeg": 65,
          "verticalAngleDeg": 135,
        },
        "processingDefinition": Map {},
        "processingMetadata": Map {
          "CHANNEL_GROUP" => "CHANNEL_GROUP",
        },
        "response": Object {
          "calibration": Object {
            "calibrationFactor": Object {
              "standardDeviation": 0.112,
              "units": "SECONDS",
              "value": 1.2,
            },
            "calibrationPeriodSec": 14.5,
            "calibrationTimeShift": "PT2S",
          },
          "effectiveAt": 1618935090.674,
          "effectiveUntil": 1618935090.674,
          "fapResponse": Object {
            "id": "966deee1-545e-44ed-a346-b029ab111151",
          },
          "id": "5dc3dce4-ae3c-3fb0-84ba-2f83fb5a89d2",
        },
        "station": Object {
          "name": "STA",
        },
        "units": "NANOMETERS",
      },
    ]
  }
  columnsToDisplay={
    Map {
      "name" => true,
      "effectiveAt" => true,
      "effectiveUntil" => true,
      "latitudeDegrees" => true,
      "longitudeDegrees" => true,
      "depthKm" => true,
      "elevationKm" => true,
      "nominalSampleRateHz" => true,
      "units" => true,
      "orientationHorizontalDegrees" => true,
      "orientationVerticalDegrees" => true,
      "calibrationFactor" => true,
      "calibrationPeriod" => true,
      "calibrationEffectiveAt" => true,
      "calibrationTimeShift" => true,
      "calibrationStandardDeviation" => true,
      "northDisplacementKm" => true,
      "eastDisplacementKm" => true,
      "verticalDisplacementKm" => true,
      "description" => true,
      "channelDataType" => false,
      "channelBandType" => false,
      "channelInstrumentType" => false,
      "channelOrientationCode" => false,
      "channelOrientationType" => false,
      "calibrationResponseId" => true,
      "fapResponseId" => true,
    }
  }
  stationData={
    Object {
      "allRawChannels": Array [
        Object {},
      ],
      "channelGroups": Array [
        Object {},
      ],
      "description": "This is a test station",
      "effectiveAt": 1618935091.118,
      "effectiveUntil": 1618935091.118,
      "location": Object {
        "depthKm": 50,
        "elevationKm": 10,
        "latitudeDegrees": 35.647,
        "longitudeDegrees": 100,
      },
      "name": "STA",
      "relativePositionsByChannel": Map {
        "Real Channel Name One" => Object {
          "eastDisplacementKm": 5,
          "northDisplacementKm": 50,
          "verticalDisplacementKm": 10,
        },
        "Real Channel Name Two" => Object {
          "eastDisplacementKm": 5,
          "northDisplacementKm": 50,
          "verticalDisplacementKm": 10,
        },
      },
      "type": "SEISMIC_1_COMPONENT",
    }
  }
>
  <div
    className="channel-configuration-table ag-theme-dark station-properties-table"
  >
    <Component
      className="bp3-heading"
    >
      <h4
        className="bp3-heading bp3-heading"
      >
        Channel Configuration
      </h4>
    </Component>
    <div
      className="station-properties-table__wrapper"
    >
      <ClientSideTable
        columnDefs={
          Array [
            Object {
              "field": "name",
              "headerName": "Name",
              "headerTooltip": "Name",
              "width": 200,
            },
            Object {
              "field": "effectiveAt",
              "headerName": "Effective At",
              "headerTooltip": "Effective at",
              "width": 200,
            },
            Object {
              "field": "effectiveUntil",
              "headerName": "Effective Until",
              "headerTooltip": "Effective until",
              "width": 200,
            },
            Object {
              "comparator": [Function],
              "field": "latitudeDegrees",
              "headerName": "Lat (째)",
              "headerTooltip": "Latitude in degrees",
            },
            Object {
              "comparator": [Function],
              "field": "longitudeDegrees",
              "headerName": "Lon (째)",
              "headerTooltip": "Longitude in degrees",
            },
            Object {
              "comparator": [Function],
              "field": "depthKm",
              "headerName": "Depth (km)",
              "headerTooltip": "Depth in kilometers",
            },
            Object {
              "comparator": [Function],
              "field": "elevationKm",
              "headerName": "Elev (km)",
              "headerTooltip": "Elevation in kilometers",
            },
            Object {
              "comparator": [Function],
              "field": "nominalSampleRateHz",
              "headerName": "Sample Rate (Hz)",
              "headerTooltip": "Nominal sample rate in hertz",
            },
            Object {
              "field": "units",
              "headerName": "Units",
              "headerTooltip": "units",
            },
            Object {
              "comparator": [Function],
              "field": "orientationHorizontalDegrees",
              "headerName": "Horiz Angle (째)",
              "headerTooltip": "Orientation horizontal angle in degrees",
            },
            Object {
              "comparator": [Function],
              "field": "orientationVerticalDegrees",
              "headerName": "Vert Angle (째)",
              "headerTooltip": "Orientation vertical angle in degrees",
            },
            Object {
              "comparator": [Function],
              "field": "calibrationFactor",
              "headerName": "Calib Factor (s)",
              "headerTooltip": "Calibration factor in seconds",
              "width": 200,
            },
            Object {
              "comparator": [Function],
              "field": "calibrationPeriod",
              "headerName": "Calib Period (s)",
              "headerTooltip": "Calibration period in seconds",
              "width": 200,
            },
            Object {
              "comparator": [Function],
              "field": "calibrationEffectiveAt",
              "headerName": "Calib Effective At",
              "headerTooltip": "Calibration effective at",
              "width": 300,
            },
            Object {
              "comparator": [Function],
              "field": "calibrationTimeShift",
              "headerName": "Calib Time Shift (s)",
              "headerTooltip": "Calibration time shift in seconds",
              "width": 200,
            },
            Object {
              "comparator": [Function],
              "field": "calibrationStandardDeviation",
              "headerName": "Calib Std Dev",
              "headerTooltip": "Calibration standard deviation",
              "width": 200,
            },
            Object {
              "comparator": [Function],
              "field": "northDisplacementKm",
              "headerName": "North Displacement (km)",
              "headerTooltip": "North displacement in kilometers",
            },
            Object {
              "comparator": [Function],
              "field": "eastDisplacementKm",
              "headerName": "East Displacement (km)",
              "headerTooltip": "East displacement in kilometers",
            },
            Object {
              "comparator": [Function],
              "field": "verticalDisplacementKm",
              "headerName": "Vert Displacement (km)",
              "headerTooltip": "Vertical displacement in kilometers",
            },
            Object {
              "field": "description",
              "headerName": "Description",
              "headerTooltip": "Description",
              "width": 300,
            },
            Object {
              "field": "channelDataType",
              "headerName": "Data Type",
              "headerTooltip": "Channel data type",
              "width": 200,
            },
            Object {
              "field": "channelBandType",
              "headerName": "Band Type",
              "headerTooltip": "Channel band type",
            },
            Object {
              "field": "channelInstrumentType",
              "headerName": "Instrument Type",
              "headerTooltip": "Channel instrument type",
            },
            Object {
              "field": "channelOrientationCode",
              "headerName": "Orientation Code",
              "headerTooltip": "Channel orientation code",
            },
            Object {
              "field": "channelOrientationType",
              "headerName": "Orientation Type",
              "headerTooltip": "Channel orientation type",
              "width": 200,
            },
            Object {
              "field": "calibrationResponseId",
              "headerName": "Calib ID",
              "headerTooltip": "Calibration response ID",
              "width": 200,
            },
            Object {
              "field": "fapResponseId",
              "headerName": "FAP Resp ID",
              "headerTooltip": "Frequency amplitude response ID",
              "width": 200,
            },
          ]
        }
        context={Object {}}
        defaultColDef={
          Object {
            "cellRendererFramework": [Function],
            "disableStaticMarkupForCellRendererFramework": true,
            "disableStaticMarkupForHeaderComponentFramework": true,
            "filter": true,
            "headerClass": "station-properties-header-cell",
            "hide": true,
            "lockVisible": false,
            "resizable": true,
            "sortable": true,
            "width": 105,
          }
        }
        deltaRowDataMode={true}
        getRowNodeId={[Function]}
        headerHeight={44}
        onGridReady={[Function]}
        overlayNoRowsTemplate="No Channels to display"
        rowData={
          Array [
            Object {
              "calibrationEffectiveAt": "2021-04-20 16:11:30.674",
              "calibrationFactor": "1.2",
              "calibrationPeriod": "14.5",
              "calibrationResponseId": "5dc3dce4-ae3c-3fb0-84ba-2f83fb5a89d2",
              "calibrationStandardDeviation": "0.112",
              "calibrationTimeShift": "2",
              "channelBandType": "B",
              "channelDataType": "Diagnostic SOH",
              "channelInstrumentType": "H",
              "channelOrientationCode": "Z",
              "channelOrientationType": "Z",
              "depthKm": "100.000",
              "description": "Example description",
              "eastDisplacementKm": "5",
              "effectiveAt": "1970-01-01 00:00:00.000",
              "effectiveUntil": "1970-01-01 00:00:00.000",
              "elevationKm": "5500.000",
              "fapResponseId": "966deee1-545e-44ed-a346-b029ab111151",
              "id": "0",
              "latitudeDegrees": "35.000",
              "longitudeDegrees": "-125.000",
              "name": "Real Channel Name One",
              "nominalSampleRateHz": "66",
              "northDisplacementKm": "50",
              "orientationHorizontalDegrees": "65",
              "orientationVerticalDegrees": "135",
              "units": "Nanometers",
              "verticalDisplacementKm": "10",
            },
          ]
        }
        rowDeselection={true}
        rowHeight={40}
        rowSelection="single"
        suppressCellSelection={true}
        suppressContextMenu={true}
      >
        <CoreTable
          columnDefs={
            Array [
              Object {
                "field": "name",
                "headerName": "Name",
                "headerTooltip": "Name",
                "width": 200,
              },
              Object {
                "field": "effectiveAt",
                "headerName": "Effective At",
                "headerTooltip": "Effective at",
                "width": 200,
              },
              Object {
                "field": "effectiveUntil",
                "headerName": "Effective Until",
                "headerTooltip": "Effective until",
                "width": 200,
              },
              Object {
                "comparator": [Function],
                "field": "latitudeDegrees",
                "headerName": "Lat (째)",
                "headerTooltip": "Latitude in degrees",
              },
              Object {
                "comparator": [Function],
                "field": "longitudeDegrees",
                "headerName": "Lon (째)",
                "headerTooltip": "Longitude in degrees",
              },
              Object {
                "comparator": [Function],
                "field": "depthKm",
                "headerName": "Depth (km)",
                "headerTooltip": "Depth in kilometers",
              },
              Object {
                "comparator": [Function],
                "field": "elevationKm",
                "headerName": "Elev (km)",
                "headerTooltip": "Elevation in kilometers",
              },
              Object {
                "comparator": [Function],
                "field": "nominalSampleRateHz",
                "headerName": "Sample Rate (Hz)",
                "headerTooltip": "Nominal sample rate in hertz",
              },
              Object {
                "field": "units",
                "headerName": "Units",
                "headerTooltip": "units",
              },
              Object {
                "comparator": [Function],
                "field": "orientationHorizontalDegrees",
                "headerName": "Horiz Angle (째)",
                "headerTooltip": "Orientation horizontal angle in degrees",
              },
              Object {
                "comparator": [Function],
                "field": "orientationVerticalDegrees",
                "headerName": "Vert Angle (째)",
                "headerTooltip": "Orientation vertical angle in degrees",
              },
              Object {
                "comparator": [Function],
                "field": "calibrationFactor",
                "headerName": "Calib Factor (s)",
                "headerTooltip": "Calibration factor in seconds",
                "width": 200,
              },
              Object {
                "comparator": [Function],
                "field": "calibrationPeriod",
                "headerName": "Calib Period (s)",
                "headerTooltip": "Calibration period in seconds",
                "width": 200,
              },
              Object {
                "comparator": [Function],
                "field": "calibrationEffectiveAt",
                "headerName": "Calib Effective At",
                "headerTooltip": "Calibration effective at",
                "width": 300,
              },
              Object {
                "comparator": [Function],
                "field": "calibrationTimeShift",
                "headerName": "Calib Time Shift (s)",
                "headerTooltip": "Calibration time shift in seconds",
                "width": 200,
              },
              Object {
                "comparator": [Function],
                "field": "calibrationStandardDeviation",
                "headerName": "Calib Std Dev",
                "headerTooltip": "Calibration standard deviation",
                "width": 200,
              },
              Object {
                "comparator": [Function],
                "field": "northDisplacementKm",
                "headerName": "North Displacement (km)",
                "headerTooltip": "North displacement in kilometers",
              },
              Object {
                "comparator": [Function],
                "field": "eastDisplacementKm",
                "headerName": "East Displacement (km)",
                "headerTooltip": "East displacement in kilometers",
              },
              Object {
                "comparator": [Function],
                "field": "verticalDisplacementKm",
                "headerName": "Vert Displacement (km)",
                "headerTooltip": "Vertical displacement in kilometers",
              },
              Object {
                "field": "description",
                "headerName": "Description",
                "headerTooltip": "Description",
                "width": 300,
              },
              Object {
                "field": "channelDataType",
                "headerName": "Data Type",
                "headerTooltip": "Channel data type",
                "width": 200,
              },
              Object {
                "field": "channelBandType",
                "headerName": "Band Type",
                "headerTooltip": "Channel band type",
              },
              Object {
                "field": "channelInstrumentType",
                "headerName": "Instrument Type",
                "headerTooltip": "Channel instrument type",
              },
              Object {
                "field": "channelOrientationCode",
                "headerName": "Orientation Code",
                "headerTooltip": "Channel orientation code",
              },
              Object {
                "field": "channelOrientationType",
                "headerName": "Orientation Type",
                "headerTooltip": "Channel orientation type",
                "width": 200,
              },
              Object {
                "field": "calibrationResponseId",
                "headerName": "Calib ID",
                "headerTooltip": "Calibration response ID",
                "width": 200,
              },
              Object {
                "field": "fapResponseId",
                "headerName": "FAP Resp ID",
                "headerTooltip": "Frequency amplitude response ID",
                "width": 200,
              },
            ]
          }
          context={Object {}}
          defaultColDef={
            Object {
              "cellRendererFramework": [Function],
              "disableStaticMarkupForCellRendererFramework": true,
              "disableStaticMarkupForHeaderComponentFramework": true,
              "filter": true,
              "headerClass": "station-properties-header-cell",
              "hide": true,
              "lockVisible": false,
              "resizable": true,
              "sortable": true,
              "width": 105,
            }
          }
          deltaRowDataMode={true}
          getRowNodeId={[Function]}
          headerHeight={44}
          onGridReady={[Function]}
          overlayNoRowsTemplate="No Channels to display"
          rowBuffer={10}
          rowData={
            Array [
              Object {
                "calibrationEffectiveAt": "2021-04-20 16:11:30.674",
                "calibrationFactor": "1.2",
                "calibrationPeriod": "14.5",
                "calibrationResponseId": "5dc3dce4-ae3c-3fb0-84ba-2f83fb5a89d2",
                "calibrationStandardDeviation": "0.112",
                "calibrationTimeShift": "2",
                "channelBandType": "B",
                "channelDataType": "Diagnostic SOH",
                "channelInstrumentType": "H",
                "channelOrientationCode": "Z",
                "channelOrientationType": "Z",
                "depthKm": "100.000",
                "description": "Example description",
                "eastDisplacementKm": "5",
                "effectiveAt": "1970-01-01 00:00:00.000",
                "effectiveUntil": "1970-01-01 00:00:00.000",
                "elevationKm": "5500.000",
                "fapResponseId": "966deee1-545e-44ed-a346-b029ab111151",
                "id": "0",
                "latitudeDegrees": "35.000",
                "longitudeDegrees": "-125.000",
                "name": "Real Channel Name One",
                "nominalSampleRateHz": "66",
                "northDisplacementKm": "50",
                "orientationHorizontalDegrees": "65",
                "orientationVerticalDegrees": "135",
                "units": "Nanometers",
                "verticalDisplacementKm": "10",
              },
            ]
          }
          rowDeselection={true}
          rowHeight={40}
          rowSelection="single"
          suppressCellSelection={true}
          suppressContextMenu={true}
        >
          <AgGridReact
            animateRows={false}
            columnDefs={
              Array [
                Object {
                  "field": "name",
                  "headerName": "Name",
                  "headerTooltip": "Name",
                  "width": 200,
                },
                Object {
                  "field": "effectiveAt",
                  "headerName": "Effective At",
                  "headerTooltip": "Effective at",
                  "width": 200,
                },
                Object {
                  "field": "effectiveUntil",
                  "headerName": "Effective Until",
                  "headerTooltip": "Effective until",
                  "width": 200,
                },
                Object {
                  "comparator": [Function],
                  "field": "latitudeDegrees",
                  "headerName": "Lat (째)",
                  "headerTooltip": "Latitude in degrees",
                },
                Object {
                  "comparator": [Function],
                  "field": "longitudeDegrees",
                  "headerName": "Lon (째)",
                  "headerTooltip": "Longitude in degrees",
                },
                Object {
                  "comparator": [Function],
                  "field": "depthKm",
                  "headerName": "Depth (km)",
                  "headerTooltip": "Depth in kilometers",
                },
                Object {
                  "comparator": [Function],
                  "field": "elevationKm",
                  "headerName": "Elev (km)",
                  "headerTooltip": "Elevation in kilometers",
                },
                Object {
                  "comparator": [Function],
                  "field": "nominalSampleRateHz",
                  "headerName": "Sample Rate (Hz)",
                  "headerTooltip": "Nominal sample rate in hertz",
                },
                Object {
                  "field": "units",
                  "headerName": "Units",
                  "headerTooltip": "units",
                },
                Object {
                  "comparator": [Function],
                  "field": "orientationHorizontalDegrees",
                  "headerName": "Horiz Angle (째)",
                  "headerTooltip": "Orientation horizontal angle in degrees",
                },
                Object {
                  "comparator": [Function],
                  "field": "orientationVerticalDegrees",
                  "headerName": "Vert Angle (째)",
                  "headerTooltip": "Orientation vertical angle in degrees",
                },
                Object {
                  "comparator": [Function],
                  "field": "calibrationFactor",
                  "headerName": "Calib Factor (s)",
                  "headerTooltip": "Calibration factor in seconds",
                  "width": 200,
                },
                Object {
                  "comparator": [Function],
                  "field": "calibrationPeriod",
                  "headerName": "Calib Period (s)",
                  "headerTooltip": "Calibration period in seconds",
                  "width": 200,
                },
                Object {
                  "comparator": [Function],
                  "field": "calibrationEffectiveAt",
                  "headerName": "Calib Effective At",
                  "headerTooltip": "Calibration effective at",
                  "width": 300,
                },
                Object {
                  "comparator": [Function],
                  "field": "calibrationTimeShift",
                  "headerName": "Calib Time Shift (s)",
                  "headerTooltip": "Calibration time shift in seconds",
                  "width": 200,
                },
                Object {
                  "comparator": [Function],
                  "field": "calibrationStandardDeviation",
                  "headerName": "Calib Std Dev",
                  "headerTooltip": "Calibration standard deviation",
                  "width": 200,
                },
                Object {
                  "comparator": [Function],
                  "field": "northDisplacementKm",
                  "headerName": "North Displacement (km)",
                  "headerTooltip": "North displacement in kilometers",
                },
                Object {
                  "comparator": [Function],
                  "field": "eastDisplacementKm",
                  "headerName": "East Displacement (km)",
                  "headerTooltip": "East displacement in kilometers",
                },
                Object {
                  "comparator": [Function],
                  "field": "verticalDisplacementKm",
                  "headerName": "Vert Displacement (km)",
                  "headerTooltip": "Vertical displacement in kilometers",
                },
                Object {
                  "field": "description",
                  "headerName": "Description",
                  "headerTooltip": "Description",
                  "width": 300,
                },
                Object {
                  "field": "channelDataType",
                  "headerName": "Data Type",
                  "headerTooltip": "Channel data type",
                  "width": 200,
                },
                Object {
                  "field": "channelBandType",
                  "headerName": "Band Type",
                  "headerTooltip": "Channel band type",
                },
                Object {
                  "field": "channelInstrumentType",
                  "headerName": "Instrument Type",
                  "headerTooltip": "Channel instrument type",
                },
                Object {
                  "field": "channelOrientationCode",
                  "headerName": "Orientation Code",
                  "headerTooltip": "Channel orientation code",
                },
                Object {
                  "field": "channelOrientationType",
                  "headerName": "Orientation Type",
                  "headerTooltip": "Channel orientation type",
                  "width": 200,
                },
                Object {
                  "field": "calibrationResponseId",
                  "headerName": "Calib ID",
                  "headerTooltip": "Calibration response ID",
                  "width": 200,
                },
                Object {
                  "field": "fapResponseId",
                  "headerName": "FAP Resp ID",
                  "headerTooltip": "Frequency amplitude response ID",
                  "width": 200,
                },
              ]
            }
            context={Object {}}
            defaultColDef={
              Object {
                "cellRendererFramework": [Function],
                "filter": true,
                "headerClass": "station-properties-header-cell",
                "hide": true,
                "lockVisible": false,
                "resizable": true,
                "sortable": true,
                "sortingOrder": Array [
                  "asc",
                  "desc",
                ],
                "suppressCellFlash": true,
                "width": 105,
              }
            }
            deltaRowDataMode={true}
            getRowNodeId={[Function]}
            headerHeight={44}
            onColumnMoved={[Function]}
            onGridReady={[Function]}
            overlayNoRowsTemplate="No Channels to display"
            preventDefaultOnContextMenu={true}
            rowBuffer={10}
            rowData={null}
            rowDataChangeDetectionStrategy="IdentityCheck"
            rowDeselection={true}
            rowHeight={40}
            rowSelection="single"
            suppressCellSelection={true}
            suppressContextMenu={true}
            suppressLoadingOverlay={true}
            suppressScrollOnNewData={true}
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </AgGridReact>
        </CoreTable>
      </ClientSideTable>
    </div>
  </div>
</ChannelConfigurationTable>
`;
